obs-studio-server/source/osn-fader.cpp

```diff
@@ -78,12 +78,13 @@ void osn::Fader::Create(
 
 	obs_fader_t* fader = obs_fader_create(type);
 	if (!fader) {
-		PRETTY_THROW("failed to create");
+		PRETTY_ERROR_RETURN(ErrorCode::Error, "Failed to create Fader.");
 	}
 
 	auto uid = Manager::GetInstance().allocate(fader);
 	if (uid == std::numeric_limits<utility::unique_id::id_t>::max()) {
-		PRETTY_THROW("invalid uid");
+		obs_fader_destroy(fader);
```

Comment:
Not sure I understand why we need to destroy the fader if we failed to allocate it

Reply:
This line from older code before PRETTY_THROW macros. 
As I understand In this case `obs_fader_create` was successful and then `allocate` failed we destroy what was created by `obs_fader_create` before return error to frontend.

Reply:
kk I see it now, sounds good 👍